django/core/management/commands/makemigrations.py

```diff
@@ -94,9 +95,18 @@ def handle(self, *app_labels, **options):
         loader = MigrationLoader(None, ignore_no_migrations=True)
 
         # Raise an error if any migrations are applied before their dependencies.
-        for db in connections:
+        consistency_check_labels = set(config.label for config in apps.get_app_configs())
+        for db in sorted(connections):
             connection = connections[db]
-            if connection.settings_dict['ENGINE'] != 'django.db.backends.dummy':
+            is_default = (db == DEFAULT_DB_ALIAS)
+            # Non-default databases are only checked if database routers used.
+            is_non_default_checkable = (db != DEFAULT_DB_ALIAS and settings.DATABASE_ROUTERS)
+            is_checkable = (
+                (is_default or is_non_default_checkable) and
+                # At least one app must be migrated to the database.
+                any(router.allow_migrate(connection.alias, label) for label in consistency_check_labels)
+            )
+            if connection.settings_dict['ENGINE'] != 'django.db.backends.dummy' and is_checkable:
```

Comment:
`ConnectionHandler` always sets up a default database, so I don't think the `if/else` is required.
https://github.com/django/django/blob/7ed8b98f42cd3a9fa530bbb764d3d79230e0a7a7/django/db/utils.py#L157-L162